src/StructureMap.DynamicInterception/MethodInvocation.cs

```diff
@@ -39,6 +54,21 @@ public MethodInfo MethodInfo
             get { return _invocation.Method; }
         }
 
+        public MethodInfo MethodInfoInvocationTarget
+        {
+            get { return _invocation.MethodInvocationTarget; }
+        }
+
+        public object Proxy
+        {
+            get { return _invocation.Proxy; }
+        }
+
+        public Type TargetType
```

Comment:
It is redundant if we provide `InvocationTarget`\`TargetObject`. One can simple call `TargetObject.GetType()`